model-optimizer/extensions/middle/EltwiseInputReshape.py

```diff
@@ -88,7 +88,7 @@ def find_and_replace_pattern(self, graph: Graph):
                     else:
                         mapping[new_shape].append(consumer)
 
-            if node.has_valid('value'):
+            if node.has_valid('value') and node.has_valid("type") and node.type == "Const":
```

Comment:
I still would like to see only the solution with adding "Reshape" operation + update the code with a Graph API instead of low-level graph modification.

Reply:
Done.